File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
answer = ''
cnt = 0
def dfs(word,q):
    global cnt, answer
    if q == word:
        answer = cnt
        return 
    if len(q) == 5:
        return 
    
    alpha = ['A','E','I','O','U']
    for i in range(5):
        cnt += 1
        q.append(alpha[i])
        dfs(word,q)
        q.pop()
    return 

def solution(word):
    word = list(word)
    q = []
    dfs(word,q)
    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>

using namespace std;

string target = "";
string aeiou = "AEIOU";
int cnt = 0;
int answer = 0;

void dfs(string cur) {
    
    if (cur == target) {
        answer = cnt;
        return;
    }
 
    if (cur.length() >= 5) {
        return;
    }
    
    for (int i = 0; i < 5; i++){
        cnt++;
        dfs(cur+aeiou[i]);
    }
}

int solution(string word) {
    target = word;
    dfs("");
    return answer;
}